src/v/pandaproxy/schema_registry/store.h

```diff
@@ -112,6 +112,37 @@ class store {
         return res;
     }
 
+    ///\brief Get the global compatibility level.
+    result<compatibility_level> get_compatibility() const {
+        return _compatibility;
+    }
+
+    ///\brief Get the compatibility level for a subject, or fallback to global.
+    result<compatibility_level> get_compatibility(const subject& sub) const {
+        auto sub_it = _subjects.find(sub);
+        if (sub_it == _subjects.end()) {
+            return error_code::subject_not_found;
+        }
+        return sub_it->second.compatibility.value_or(_compatibility);
+    }
+
+    ///\brief Set the global compatibility level.
+    result<bool> set_compatibility(compatibility_level compatibility) {
+        return std::exchange(_compatibility, compatibility) != compatibility;
+    }
+
+    ///\brief Set the compatibility level for a subject.
+    result<bool>
+    set_compatibility(const subject& sub, compatibility_level compatibility) {
+        auto sub_it = _subjects.find(sub);
+        if (sub_it == _subjects.end()) {
+            return error_code::subject_not_found;
+        }
+        // TODO(Ben): Check needs to be made here?
+        return std::exchange(sub_it->second.compatibility, compatibility)
+               != compatibility;
```

Comment:
> is there any guarantee that `sub_it->second.compatibility` will not be std::nullopt? if it std::nullop I always get confused by the semantics of the default std::optional operator== in this case. I guess if it is nullopt it certainly wouldn't be equal to `compatibility` so that seems to make sense.

It starts as `nullopt`, so it's a pretty common case. Maybe the return value is wrong if it's set from `nullopt` to `nullopt`, but that's not possible here. A later commit (in a subsequent PR that doesn't exist yet) introduces an explicit `clear_compatibility` method on the store.